Winget-AutoUpdate/functions/Test-ListPath.ps1

```diff
@@ -0,0 +1,67 @@
+#Function to check Black/White List External Path
+
+function Test-ListPath ($ListPath, $UseWhiteList, $WingetUpdatePath) {
+    # UNC, Web or Local Path
+    if ($UseWhiteList){
+        $ListType="included"
+    }
+    else {
+        $ListType="excluded"
+    }
+    $LocalList = -join($WingetUpdatePath, "\", $ListType, "_apps.txt")
+    if (Test-Path "$LocalList") {
+        $dateLocal = (Get-Item "$LocalList").LastWriteTime.ToString("yyyy-MM-dd HH:mm:ss")
+    }
+    $ExternalList = -join($ListPath, "\", $ListType, "_apps.txt")
+    $PathInfo=[System.Uri]$ListPath
+
+    if($PathInfo.IsUnc){
```

Comment:
This part is not the same as the last else ?

Reply:
This $ExternalList is defined above the If - elseif - else and changed in in elseif (http*) because it's the same in an UNC/Local path

Reply:
but something like this could not be good ?
```
# Function to check Black/White List External Path

function Test-ListPath ($ListPath, $UseWhiteList, $WingetUpdatePath) {
	# UNC, Web or Local Path
	if ($UseWhiteList){
		$ListType="included_apps.txt"
	}
	else {
		$ListType="excluded_apps.txt"
	}

	# Get local and external list paths
	$LocalList = -join($WingetUpdatePath, "\", $ListType)
	$ExternalList = -join($ListPath, "\", $ListType)

	# Check if a list exists
	if (Test-Path "$LocalList") {
		$dateLocal = (Get-Item "$LocalList").LastWriteTime.ToString("yyyy-MM-dd HH:mm:ss")
	}

	# If path is URL
	if ($ListPath -like "http*"){
		$ExternalList = -join($ListPath, "/", $ListType)
		$wc = New-Object System.Net.WebClient
		try {
			$wc.OpenRead("$ExternalList").Close() | Out-Null
			$dateExternal = ([DateTime]$wc.ResponseHeaders['Last-Modified']).ToString("yyyy-MM-dd HH:mm:ss")
			if ($dateExternal -gt $dateLocal) {
				try {
					$wc.DownloadFile($ExternalList, $LocalList)
				}
				catch {
					return $False
				}
				return $true
			}
		}
		catch {
			return $False
		}
	}
	# If path is UNC or local
	else {
		if(Test-Path -Path $ExternalList -PathType leaf){
			$dateExternal = (Get-Item "$ExternalList").LastWriteTime.ToString("yyyy-MM-dd HH:mm:ss")
			if ($dateExternal -gt $dateLocal) {
				try {
					Copy-Item $ExternalList -Destination $LocalList -Force
				}
				catch {
					return $False
				}
				return $true
			}
		}
	}
	return $false
}
```

Reply:
Your're the BOSS ;)